contracts/NumeraireDelegateV4.sol

```diff
@@ -0,0 +1,183 @@
+pragma solidity >=0.4.11 <0.5.0;
+
+import "contracts/StoppableShareable.sol";
+import "contracts/NumeraireShared.sol";
+
+// Whoever creates the contract has the power to stop it, this person can be changed via transferOwnership(_new_address)
+contract NumeraireDelegateV2 is StoppableShareable, NumeraireShared {
+
+    function NumeraireDelegateV2(address[] _owners, uint256 _num_required) StoppableShareable(_owners, _num_required) {
+    }
+
+    // All minted NMR are initially sent to Numerai, obeying both weekly and total supply caps
+    function mint(uint256 _value) onlyOwner returns (bool ok) {
+        // Prevent minting more than the supply cap.
+        require(safeAdd(total_minted, _value) <= supply_cap);
+
+        // Prevent minting more than the disbursement.
+        require(_value <= getMintable());
+
+        balanceOf[numerai] = safeAdd(balanceOf[numerai], _value);
+        totalSupply = safeAdd(totalSupply, _value);
+        total_minted = safeAdd(total_minted, _value);
+
+        // Notify anyone listening.
+        Mint(_value);
+
+        return true;
+    }
+
+    // Numerai calls this function to release staked tokens when the staked predictions were successful
+    function releaseStake(address _staker, bytes32 _tag, uint256 _burnAmt, uint256 _tournamentID, uint256 _roundID, bool _successful) onlyOwner stopInEmergency returns (bool ok) {
+        var round = tournaments[_tournamentID].rounds[_roundID];
+        var stake = round.stakes[_staker][_tag];
+        var originalStakeAmount = stake.amount;
+
+        require(stake.amount > 0);
+        require(!stake.resolved);
+        require(round.resolutionTime <= block.timestamp);
+
+        stake.amount = 0;
+        balanceOf[_staker] = safeAdd(balanceOf[_staker], safeSubtract(originalStakeAmount, _burnAmt));
+        stake.resolved = true;
+        stake.successful = _successful;
+
+        totalSupply = safeSubtract(totalSupply, _burnAmt);
+
+        StakeReleased(_tournamentID, _roundID, _staker, _tag, _burnAmt);
+        StakeDestroyed(_tournamentID, _roundID, _staker, _tag);
```

Comment:
do we have to emit this event?

what if we just emitted the stakeReleased event?

Reply:
addressed